### ISSUE TITLE
Rule proposal: `prefer-switch`

### ISSUE THREAD

**Issue by Torhal** [sindresorhus/eslint-plugin-unicorn#1051]
A switch statement is easier to read than multiple if statements with simple equality comparisons.

## Fail

```js
if (foo === bar) {
    /* ... */
}
else if (foo === baz) {
    /* ... */
}
else if (foo === arglebargl) {
    /* ... */
}
else if (foo === glopglyph) {
    /* ... */
}
```

## Pass

```js
if (foo === bar) {
    /* ... */
}
else if (foo === baz) {
    /* ... */
}
```

```js
switch (foo) {
    case bar:
        /* ... */
        break;
    case baz:
        /* ... */
        break;
    case arglebargl:
        /* ... */
        break;
    case glopglyph:
        /* ... */
        break;
}
```
This could have a ```minimumCases``` option, defaulting to 3.

**Comment by sindresorhus** [comment]
> This could have a minimumCases option, defaulting to 3.

👍

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by arpowers** [comment]
I don't understand why people like `switch` statements.  This rule is very much up for debate.
